feature_engine/transformation/log.py

```diff
@@ -164,3 +164,206 @@ def _more_tags(self):
         tags_dict["_xfail_checks"]["check_transformer_general"] = msg
 
         return tags_dict
+
+
+class LogCpTransformer(BaseNumericalTransformer):
+    """
+    The LogCpTransformer() applies the transformation log(x + C), where C is a positive
+    constant, to the input variable. It applies the natural logarithm or the base 10
+    logarithm, where the natural logarithm is logarithm in base e.
+
+    The LogCpTransformer() only works with numerical non-negative values. If the
+    variable contains a zero or a negative value after adding a constant C, the
+    transformer will return an error.
+
+    A list of variables can be passed as an argument. Alternatively, the transformer
+    will automatically select and transform all variables of type numeric.
+
+    Parameters
+    ----------
+    variables: list, default=None
+        The list of numerical variables to transform. If None, the transformer
+        will find and select all numerical variables.
+
+    base: string, default='e'
+        Indicates if the natural or base 10 logarithm should be applied. Can take
+        values 'e' or '10'.
+
+    C: "auto" or int, default="auto"
+        The constant C to apply the transfomation log(x + C).
+
+        - If int, then log(x + C)
+        - If "auto", then C = abs(min(x)) + 1
+
+    Attributes
+    ----------
+    variables_:
+        The group of variables that will be transformed.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    C_:
+        The constant C to transform data.
+
+    Methods
+    -------
+    fit:
+        This transformer learns the constant C to add to log transformation.
+    transform:
+        Transforms the variables using log transformation.
+    fit_transform:
+        Fit to data, then transform it.
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        base: str = "e",
+        C: Union[int, str] = "auto",
+    ) -> None:
+
+        if base not in ["e", "10"]:
+            raise ValueError("base can take only '10' or 'e' as values")
+
+        if not isinstance(C, int) and not C == "auto":
+            raise ValueError("C can take only 'auto' or int")
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.base = base
+        self.C = C
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer learns the constant C to add to log transformation.
```

Comment:
**If auto,** this transformer learns the constant C to add to log transformation.

Could we add the above? otherwise it might be confusing?